zilliqa/src/block_store.rs

```diff
@@ -122,9 +150,22 @@ impl BlockStore {
             self.requested_view = current_view;
         }
 
+        // Attempt to send pending requests if we can.
+        while let Some((from, to)) = self.pending_requests.pop() {
+            if !self.request_blocks(from, to)? {
+                // Stop trying to send requests if no peers are available.
+                break;
+            }
```

Comment:
How do we cope if no peers are available for an extended period and self.pending_requests starts to get big?
I guess it probably doesn't matter too much, since we will need all these blocks in the order in which they appear in the list in any event, and so pruning it from either end will do us no good. We might consider (perhaps a comment here?) throttling the number of these we send at once to avoid becoming a noisy neighbour when the network comes back and we have a long list of self.pending_requests...

Reply:
The length of `self.unserviceable_requests` is (implicitly) bounded by `self.max_blocks_in_flight`.

I do agree that being a noisy neighbour might be a problem - If we learn about a single node at first, that poor guy will have to service all of our requests. Will raise an issue on this. I think this should probably be based on changing `self.max_blocks_in_flight` to a per-peer limit instead?